src/ape/api/config.py

```diff
@@ -425,12 +434,30 @@ def deployments(self) -> dict[str, dict[str, list[DeploymentConfig]]]:
                 raise ConfigError(f"Invalid ecosystem '{ecosystem_name}' in deployments config.")
 
             ecosystem = self.network_manager.ecosystems[ecosystem_name]
-            for network_name, network_deploys in ecosystem_deploys.items():
+            for network_name in ecosystem_deploys:
                 if network_name not in ecosystem.networks:
                     raise ConfigError(
                         f"Invalid network '{ecosystem_name}:{network_name}' in deployments config."
                     )
 
+        # NOTE: Merge deterministic deployments after we initializing `NetworkManager.ecosystems`
+        for ecosystem_name, ecosystem in self.network_manager.ecosystems.items():
+            if ecosystem_name not in self.deployment_data:
```

Comment:
you can do this neat trick:

```
self.deployment_data.setdefault(ecosystem_name, {})
```

It does exactly the same as what you're doing here:

```
for ecosystem_name, ecosystem in self.network_manager.ecosystems.items():
            if ecosystem_name not in self.deployment_data:
```